sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -184,14 +186,20 @@ impl CAILoader for JpegIO {
                                 buffer.append(&mut raw_vec.as_mut_slice()[16..].to_vec());
 
                                 cai_seg_cnt += 1;
-                            } else {
+                            } else if raw_vec.len() > 28 {
```

Comment:
These magic numbers are kind of scary, maybe add some comments about where these numbers come from

Reply:
There is no magic number.  The data starts after the header.